.github/workflows/publish.yml

```
@@ -0,0 +1,21 @@
+name: Publish NuGet Package
+
+on:
+  push:
+    tags:
+      - 'v*.*.*.*'  # Trigger only on version tags like v1.2.3.4
```

Instead of having `publish-dry-run` let's make it possible to trigger it manually:
```suggestion
on:
  push:
    tags:
      - 'v*.*.*.*'  # Trigger only on version tags like v1.2.3.4
  workflow_dispatch:
    inputs:
      target_tag:
        description: "Publish particular tag"
        type: string
        required: false
        default: ""

      dry-run:
        description: "Publish to a dry-run package"
        type: boolean
        required: true
        default: true
```